packages/vm/src/vmState.ts

```diff
@@ -0,0 +1,518 @@
+import Common from '@ethereumjs/common'
+import { AccessList, AccessListItem } from '@ethereumjs/tx'
+import { Account, Address, toBuffer } from 'ethereumjs-util'
+const Set = require('core-js-pure/es/set')
+
+import { StateManager } from './state/index'
+import { AccountFields } from './state/interface'
+
+import { getActivePrecompiles, ripemdPrecompileAddress } from './evm/precompiles'
+import { debug as createDebugLogger, Debugger } from 'debug'
+
+type AddressHex = string
+
+interface BaseContext {
+  touchAccount(address: Address): void
+  checkpoint(): Promise<void>
+  commit(): Promise<void>
+  revert(): Promise<void>
+}
+
+export interface EIP2929Context extends BaseContext {
+  addWarmedAddress(address: Buffer): void
+  isWarmedAddress(address: Buffer): boolean
+  addWarmedStorage(address: Buffer, slot: Buffer): void
+  isWarmedStorage(address: Buffer, slot: Buffer): boolean
+  clearWarmedAccounts(): void
+  generateAccessList?(addressesRemoved: Address[], addressesOnlyStorage: Address[]): AccessList
+  getOriginalContractStorage(address: Address, key: Buffer): Promise<Buffer>
+  clearOriginalStorageCache(): void
+  cleanupTouchedAccounts(): Promise<void>
+
+  getAccount(address: Address): Promise<Account>
+  putAccount(address: Address, account: Account): Promise<void>
+  deleteAccount(address: Address): Promise<void>
+}
```

Comment:
1. You can throw this `EIP2929Context` together with the `BaseContext` as some simplification (and then maybe call this `StateContext` or similar as something more expressive?), this is on the list of simplifications for the breaking releases anyhow.

Reply:
2. I guess `getAccount`, `putAccount` and `deleteAccount` don't belong here? 🤔

Reply:
I think yes what you are saying has merit. i think, we should have `vm.getAccount`, `vm.putAccount`, `vm.deleteAccount` which combines the statemanager call with maintaining touched addresses in vmState. what do you think?

Reply:
Ah, yes, that makes sense, didn't saw the broader context, thanks for the explanation! 👍